src/utils/formatting/numberFormatting.js

```diff
@@ -1,5 +1,9 @@
 /* Takes a number or string */
 export const addCommas = number => {
+  // must check array and boolean separately they both are not NaN
```

Comment:
I think there are some cases missed with this approach.  e.g. `isNaN(null)` is false.  My guess is that your 'should throw an error if given null' test passed not because a TypeError was thrown on line 5, but because a null pointer exception occurred on line 7.

A more bulletproof approach might be:
```
export const addCommas = number => {
  const parsedNum = parseFloat(number);
  if (isNaN(parsedNum)) {
    throw new TypeError(`${addCommas.name} requires a numeric input or input that can be parsed to a number.`);
  }
  let parts = parsedNum().split('.');
  ...
}
```

Ran `node` and played around with `parseFloat` a bit in a terminal:
<img width="344" alt="screen shot 2018-01-28 at 1 02 19 pm" src="https://user-images.githubusercontent.com/9269522/35487014-862d2478-042b-11e8-96c1-695de576cefe.png">